tests/errors/restriction_error_01.py

```diff
@@ -0,0 +1,6 @@
+from ltypes import TypeVar, Any
+
+T = TypeVar('T', bound=Any)
+
+def f(x: T, y: T) -> T:
+    return x + y
```

Comment:
We need to brainstorm the name for this --- and whether to even allow it at all.

I believe in Rust (or Go?) "Any" means "any type" can be used, which means you are not allowed to do almost anything on it, specifically you are not allowed to do "a+b", because that does not work with any type, only with types that support it.

Also, how is this checked --- it seems that here we would still need to go into the template to check that the type can be used? 

It seems to me we should not allow this at all. That is, "Any" could mean "any user type will work here", i.e., "no restrictions", but then you can't use "+" in the function, so the above should not compile. You would have to specify restrictions here.

Reply:
Yes, that's an example of a restriction error. It wouldn't pass the type check. Maybe the name Any is a bit inappropriate and Assignable would be better for it.

The check is very simple, the binary operation x + y would require both x and y to have SupportsPlus in their bounds if they are type parameters.

Reply:
Ah, I forgot that this is an "error", this is good. There is no problem.